drivers/regex/mlx5/mlx5_regex.h

```diff
@@ -35,6 +35,8 @@ struct mlx5_regex_priv {
 	uint16_t nb_queues; /* Number of queues. */
 	struct mlx5_regex_qp *qps; /* Pointer to the qp array. */
 	uint16_t nb_max_matches; /* Max number of matches. */
+	//ORI check, where best to add this mode? :
+	enum mlx5_rxp_program_mode prog_mode;
 	struct mlx5_regex_db db[MLX5_RXP_MAX_ENGINES +
 				MLX5_RXP_SHADOW_EM_COUNT];
```

Comment:
do we need the extra memory MLX5_RXP_SHADOW_EM_COUNT

Reply:
This memory was added to speed up programming rules originally, I'd suggest leaving it until we can validate more programming sequences to see if it can be of benefit.  For example this would allow as to preprogram EM rules before taking RXP engine off line (in an already running system).  Although this has not be validated as we have only ever programmed 2 memories todate!

Reply:
O.K